src/hipscat/pixel_math/margin_bounding.py

```diff
@@ -0,0 +1,172 @@
+        for a given margin bounding box scale.
+
+    Args:
+        pixel_order (int): the order of the pixel to which we're calculating the margin region for.
+        pix (int): the healpixel which we're calculating the margin region for.
+        scale (float): the scale factor to apply to the given healpixel.
+        step (int): the amount of samples of one side of the given healpixel's boundaries.
+            total samples taken = 4 * step.
+    Returns:
+        polygons (list of tuples): a list of obj:`regions.PolygonPixelRegion` and obj:`wcs.WCS`
+            tuples, covering the full area of the given healpixels scaled up area.
+    """
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-statements
+
+    pixel_boundaries = hp.vec2dir(
+        hp.boundaries(2**pixel_order, pix, step=step, nest=True), lonlat=True
+    )
+
+    # find the translation values to keep the bounding box
+    # centered around the orignal healpixel.
+    n_samples = len(pixel_boundaries[0])
+    centroid_lon = np.sum(pixel_boundaries[0]) / n_samples
+    centroid_lat = np.sum(pixel_boundaries[1]) / n_samples
+    translate_lon = centroid_lon - (centroid_lon * scale)
+    translate_lat = centroid_lat - (centroid_lat * scale)
+
+    affine_matrix = np.array(
+        [[scale, 0, translate_lon], [0, scale, translate_lat], [0, 0, 1]]
+    )
+
+    # convert the orignal boundary coordinates into
+    # a homogenous coordinate space (3-dim)
+    homogeneous = np.ones((3, n_samples))
+    homogeneous[0] = pixel_boundaries[0]
+    homogeneous[1] = pixel_boundaries[1]
+
+    transformed_bounding_box = np.matmul(affine_matrix, homogeneous)
+
+    # if the transform places the declination of any points outside of
+    # the range 90 > dec > -90, change it to a proper dec value.
+    for i in range(len(transformed_bounding_box[1])):
+        dec = transformed_bounding_box[1][i]
+        if dec > 90.0:
+            transformed_bounding_box[1][i] = 90.0 - (dec - 90.0)
+        elif dec < -90.0:
+            transformed_bounding_box[1][i] = -90.0 - (dec + 90.0)
+
+    min_ra = np.min(transformed_bounding_box[0])
+    max_ra = np.max(transformed_bounding_box[0])
+    min_dec = np.min(transformed_bounding_box[1])
+    max_dec = np.max(transformed_bounding_box[1])
+
+    # one arcsecond
+    pix_size = 0.0002777777778
+
+    ra_naxis = int((max_ra - min_ra) / pix_size)
+    dec_naxis = int((max_dec - min_dec) / pix_size)
+
+    polygons = []
+    # for k < 2, the size of the bounding boxes breaks the regions
+    # code, so we subdivide the pixel into multiple parts with
+    # independent wcs.
+    if pixel_order < 2:
+        # k == 0 -> 16 subdivisions, k == 1 -> 4 subdivisions
+        divs = 4 ** (2 - pixel_order)
+        div_len = int((step * 4) / divs)
+        for i in range(divs):
+            j = i * div_len
+            lon = list(transformed_bounding_box[0][j : j + div_len + 1])
```

Comment:
for indexing we're dividing up the boundary values by the number of subdivisions we're going to use, then using that subset in addition to the centroid to create a slice of the bounding box. the main tricky thing here is that we want to include the first point of the next slice in each subdivision so that we don't have any gaps, which is why we end the index at div_len + 1.
so for instance, when we want to create 4 subdivisions on a step = 10 bounding box (40 total samples)
```
i = 0
j = 0
transformed_bounding_box[0][0 : 11] -> 11 values,  indexes 0 through 10
...
i = 1
j = 10
transformed_bounding_box[0][10 : 21] -> 11 values, indexes 10 through 20
```

I can add a clarifying comment to make it's clear what it does.